contracts/protocol/libraries/logic/BorrowLogic.sol

```diff
@@ -99,9 +101,21 @@ library BorrowLogic {
         );
 
         if (params.releaseUnderlying) {
+            DataTypes.TimeLockParams memory timeLockParams = GenericLogic
+                .calculateTimeLockParams(
+                    reserve,
+                    DataTypes.TimeLockFactorParams({
+                        assetType: DataTypes.AssetType.ERC20,
+                        asset: params.asset,
+                        amount: params.amount
```

Comment:
We should not using params.amout here because -1 (i.e. uint256.max)  is possiblely passed in for denoting withdraw all. In this case locked time returned by `calculateTimeLockParams ` might be wrong.

Reply:
Also it will cause `calculateTimeLockParams` revert because there is a explicit conversion from u256 to u128
```
        timeLockParams.releaseTime +=
            currentTimestamp +
            _updatePeriodLimit(currentTimestamp, uint128(params.amount));
```